### PR TITLE
fixing BC serialization bug with nnq.dropout

### PR DESCRIPTION
**Pull request by HDCharles**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* __->__ #71430

to fix https://github.com/pytorch/vision/issues/5198

problem was that nnq.dropout was supposed to lose its post_activation_process upon from_float being called, but for qat the module was being converted at the wrong point. This changed the state_dict and prevented it from being BC.

Differential Revision: [D33640816](https://our.internmc.facebook.com/intern/diff/D33640816/)

### CODE REVIEW

**Review by jerryzh168 — APPROVED** [review]

**Review by terrychenism — APPROVED** [review]

**Review by datumbox — APPROVED** [review]

**Comment by datumbox** [comment]
Thanks for looking into this. If additional investigation is needed to fix Dropout or more time is required to merge this PR, I would appreciate it if we revert the original PR that broke TorchVision's CI as it's making our development harder.
